### DISCUSSION TITLE
Manually renew IP address without restarting the container

### DISCUSSION THREAD

**Discussion by seth100** [qdm12/gluetun#1593] [Q&A] [answered]
Hi, 

I am using gluetun in a docker compose stack running other containers connected. My VPN provider is Mullvad with multiple server countries and I use the Wireguard protocol. 
Is there a way to manually renew (obtain a new) IP address without restarting the gluetun container?
Thanks

**Comment by seth100** [comment]
Anyone? There is really no way to do that?

**Comment by DcR-NL** [comment, marked as answer]
You can stop and start the VPN using the [Control Server](https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/control-server.md). It also allows you to query your public IP.

For example, using curl:
```
# curl -X PUT --fail --silent --show-error http://localhost:8000/v1/openvpn/status -H "Content-Type: application/json" -d '{"status":"stopped"}'
{"outcome":"stopped"}
# curl -X PUT --fail --silent --show-error http://localhost:8000/v1/openvpn/status -H "Content-Type: application/json" -d '{"status":"running"}'
{"outcome":"running"}
```

**Reply by seth100** [reply]
great thanks!

**Comment by SR-G** [comment]
Works fine, for reference, for wireguard, it should probably be now something like (seems "/vpn" endpoints are now exposed) : 

```bash
URL="https://<basicauth, ...>@gluetun-api.domain"
curl -X PUT --fail --silent --show-error ${URL}/v1/vpn/status -H "Content-Type: application/json" -d '{"status":"stopped"}'
curl -X PUT --fail --silent --show-error ${URL}/v1/vpn/status -H "Content-Type: application/json" -d '{"status":"running"}'
```
